### PR TITLE
JAMES-3946 DropLists

### PR DESCRIPTION
**Pull request by Maxxx873**
[https://issues.apache.org/jira/projects/JAMES/issues/JAMES-3946](https://issues.apache.org/jira/projects/JAMES/issues/JAMES-3946?filter=allopenissues)

### CODE REVIEW

File: server/data/data-api/src/main/java/org/apache/james/droplists/api/DropListEntry.java [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+        return deniedEntityType;
+    }
+
+    public Optional<String> getDeniedEntity() {
+        return deniedEntity;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
```
Comment (chibenwa) [inline server/data/data-api/src/main/java/org/apache/james/droplists/api/DropListEntry.java:105]: Please write a POJO test with EqualsVelifier. I bet this POJO is non-compliant...

File: server/data/data-api/src/main/java/org/apache/james/droplists/api/DropListEntry.java [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+        public DropListEntry build() {
+            return new DropListEntry(ownerScope, owner, deniedEntityType, deniedEntity);
+        }
+    }
+
+    private Optional<OwnerScope> ownerScope;
+    private Optional<String> owner;
+    private Optional<String> deniedEntityType;
+    private Optional<String> deniedEntity;
```
Comment (chibenwa) [inline server/data/data-api/src/main/java/org/apache/james/droplists/api/DropListEntry.java:78]: Aren't those all mandatory?

Can we really have a drop list entry without a target entity?

File: server/data/data-api/src/main/java/org/apache/james/droplists/api/DropListEntry.java [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+        }
+
+        public DropListEntry build() {
+            return new DropListEntry(ownerScope, owner, deniedEntityType, deniedEntity);
+        }
+    }
+
+    private Optional<OwnerScope> ownerScope;
+    private Optional<String> owner;
+    private Optional<String> deniedEntityType;
```
Comment (chibenwa) [inline server/data/data-api/src/main/java/org/apache/james/droplists/api/DropListEntry.java:77]: We shall have an enum for deniedEntityType; IMO

**Review by chibenwa — COMMENTED** [review]
Thanks!

As this only is an API, and as we will need storage implementations, webadmin utilities and mailet tooling to make sense of this, I propose not to merge this work  too early.

Maybe we can set up a feature branch?

Also, I have a couple of cosmetic remarks.

**Comment by Maxxx873** [comment]
> Thanks!
> 
> As this only is an API, and as we will need storage implementations, webadmin utilities and mailet tooling to make sense of this, I propose not to merge this work too early.
> 
> Maybe we can set up a feature branch?
> 
> Also, I have a couple of cosmetic remarks.

Thanks for the helpful remarks! Yes, I think when you create a new feature branch I will edit the PR.

**Comment by chibenwa** [comment]
Here you are: https://github.com/apache/james-project/tree/droplist

**Review by Arsnael — APPROVED** [review]

**Review by chibenwa — APPROVED** [review]

**Comment by Arsnael** [comment]
Thanks for the api contribution for this :) I merge this in the feature branch

**Pull request merged**